pkg/controller/datavolume/controller.go

```diff
@@ -842,3 +846,52 @@ func (r *ReconcilerBase) shouldBeMarkedWaitForFirstConsumer(pvc *corev1.Persiste
 
 	return res, nil
 }
+
+// handlePvcCreation works as a wrapper for non-clone PVC creation and error handling
+func (r *ReconcilerBase) handlePvcCreation(log logr.Logger, syncRes *dataVolumeSyncResult,
+	pvcModifier func(datavolume *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim)) error {
+	if _, dvPrePopulated := syncRes.dv.Annotations[cc.AnnPrePopulated]; dvPrePopulated {
+		return nil
+	}
+	if syncRes.pvc != nil {
+		return nil
+	}
+	// Creating the PVC
+	newPvc, err := r.createPvcForDatavolume(syncRes.dv, syncRes.pvcSpec, pvcModifier)
+	if err != nil {
+		if cc.ErrQuotaExceeded(err) {
+			r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Pending, syncRes.dv, nil, nil,
```

Comment:
It will need to be change to `syncDataVolumeStatusPhaseWithEvent()` after [this](https://github.com/kubevirt/containerized-data-importer/pull/2525) gets in, as we won't allow status update from `sync()`.

Reply:
PopulatorReconciler also needs to be adapted to the new sync/updateStatus model, so please look at the [PR](https://github.com/kubevirt/containerized-data-importer/pull/2525) changes in the other DV controllers.

Reply:
Thanks for sharing, already working on the changes :+1: